scripts/src/admin/translations.ts

```diff
@@ -266,37 +275,45 @@ async function insertIntoDb(data: OffersTranslation, language: string) {
 	});
 	console.info(`inserting ${ dbReadyData.length } translations`);
 	await OfferTranslation.createQueryBuilder().insert().values(dbReadyData)
-		.onConflict(`("offer_id", "context", "path", "language") DO NOTHING`)
+		.onConflict(`("offer_id", "context", "path", "language") DO UPDATE SET offer_id=EXCLUDED.offer_id, context=EXCLUDED.context, path=EXCLUDED.path, language=EXCLUDED.language`)
 		.execute();
 }
 
 //  TODO: add validation
 async function processTranslationData(csvDataRows: TranslationData) {
+	const allOffers = await Offer.find({ type: "earn" });
 	const allOfferContents = await OfferContent.find({ select: ["offerId", "content"] } as FindManyOptions<OfferContent>);
 	const allContentTranslations: OffersTranslation = {};
 	csvDataRows.forEach(([__, csvKey, ___, translation]) => {
 		if (!translation) {
 			return;
 		}
-		const [table, offerId, column, jsonPath] = getCsvKeyElements(csvKey);
-		let offerTranslations;
-		if (offerId in allContentTranslations) {
-			offerTranslations = allContentTranslations[offerId];
-		} else {
-			offerTranslations = { content: getOfferContentFromJson(allOfferContents.find(content => content.offerId === offerId)) } as OfferTranslationData;
-		}
-		if (table === "offer") {
-			offerTranslations[column] = translation;
-		} else {
-			const evalString = `offerTranslations.content.${ jsonPath }=translation`;
-			try {
-				/* tslint:disable-next-line:no-eval */
-				eval(evalString);
-			} catch (e) {
-				console.error("content eval failed: \neval string: %s\n error message: %s", evalString, e);
+		// const [table, offerId, column, jsonPath] = getCsvKeyElements(csvKey);
+		const [table, offerName, column, jsonPath] = getCsvKeyElements(csvKey);
+		const currentOffer = _.find(allOffers, function(offer: Offer) {
```

Comment:
Why lodash and not just `allOffers.find()`?

If you are doing this just to get the offer id in-order to get to the offer content, you can just do one query and join the two tables together.

Reply:
Also use arrow functions instead of function definistions.

Reply:
using js find instead. hard to get used to *not* using _